.github/workflows/ci.yml

```
@@ -1,157 +1,276 @@
+          cache: 'pnpm'
+
+      - name: 📚 Install dependencies
+        run: pnpm install --frozen-lockfile
+
+      - name: 🔍 Run security audit
+        run: |
+          pnpm audit --audit-level moderate
+          npx better-npm-audit audit --level moderate
+
+      - name: 🛡️ Run CodeQL Analysis
+        uses: github/codeql-action/analyze@v2
```

actionlint (1.7.7)

the runner of "github/codeql-action/analyze@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)